src/affinity.cc

```diff
@@ -0,0 +1,89 @@
+// non-platform-specific code for thread pinning
+
+#include "affinity.h"
+#include "log.h"
+
+#include <cstdint>
+#include <ranges>
+#include <string_view>
+#include <vector>
+
+#include <libenvpp/env.hpp>
+
+namespace celerity::detail::thread_pinning {
+
+std::string thread_type_to_string(const thread_type t_type) {
+	switch(t_type) {
+	case thread_type::user: return "user";
+	case thread_type::scheduler: return "scheduler";
+	case thread_type::executor: return "executor";
+	default: break;
+	}
+	if(t_type >= thread_type::first_backend_worker && t_type < thread_type::first_host_queue) {
+		return fmt::format("backend_worker_{}", t_type - thread_type::first_backend_worker);
+	}
+	if(t_type >= thread_type::first_host_queue) { return fmt::format("host_queue_{}", t_type - thread_type::first_host_queue); }
+	return "unknown";
+}
+
+namespace {
+	// When we no longer need to support compilers without a working std::views::split, get rid of this function
+	std::vector<std::string> split(const std::string_view str, const char delim) {
+		std::vector<std::string> result;
+		size_t start = 0;
+		size_t end = str.find(delim);
+		while(end != std::string::npos) {
+			result.push_back(std::string(str.substr(start, end - start)));
+			start = end + 1;
+			end = str.find(delim, start);
+		}
+		if(start < str.size()) result.push_back(std::string(str.substr(start)));
+		return result;
+	}
+} // namespace
+
+environment_configuration parse_validate_env(const std::string_view str) {
+	using namespace std::string_view_literals;
+	constexpr const char* error_msg =
+	    "Cannot parse CELERITY_THREAD_PINNING setting, needs to be either 'auto', 'from:#', comma-separated core list, or bool: {}";
```

Comment:
Just a note that we should document the core-list somewhere, i.e. which index is which thread and also how this interacts with legacy mode.

Reply:
Done in the upcoming documentation overhaul. However, since this is a very low-level feature I didn't duplicate the information in the docs, since it would probably just get outdated. I rather just point to the relevant code.